pymc3/model.py

```diff
@@ -1518,3 +1472,343 @@ def all_continuous(vars):
+    """Performs a breadth first search on the supplied root node's logpt or
+    transformed logpt graph searching for named input nodes, which are
+    different from the supplied root. Each explored branch will stop when
+    either when it ends or when it finds its first named node.
+
+    Parameters
+    ----------
+    root: theano.Variable (mandatory)
+        The node from which to get the transformed.logpt or logpt and perform
+        the search. If root does not have either of these attributes, the
+        function returns None.
+
+    Returns
+    -------
+    conditional_on : set, with named nodes that are not theano.Constant nor
+    SharedVariable. The input `root` is conditionally dependent on these nodes
+    and is one step away from them in the bayesian network that specifies the
+    relationships, hence the name `get_first_level_conditionals`.
+    """
+    transformed = getattr(root, 'transformed', None)
+    try:
+        cond = transformed.logpt
+    except AttributeError:
+        cond = getattr(root, 'logpt', None)
+    if cond is None:
+        return None
+    conditional_on = set()
+    queue = copy(getattr(cond.owner, 'inputs', []))
+    while queue:
+        parent = queue.pop(0)
+        if (parent is not None and getattr(parent, 'name', None) is not None
+                and not_shared_or_constant_variable(parent)):
+            # We don't include as a conditional relation either logpt depending
+            # on root or on transformed because they are both deterministic
+            # relations
+            if parent == root or parent == transformed:
+                continue
+            # Some theano ops place default names to their associated output
+            # nodes. We want to ignore them, so we test that the name pattern
+            # matches with the ones we don't want, that the op is an instance
+            # of the conflicting op, and, in the case of mean, var, and std,
+            # where the op is built from other ops, that the op.name matches
+            # the default one.
+            must_ignore = False
+            for name_pattern, owner_type, owner_name in _theano_autonamed_ops:
+                if re.match(name_pattern, parent.name):
+                    owner = getattr(parent, 'owner', None)
+                    if owner is not None and isinstance(owner.op, owner_type):
+                        if (owner_name is None or
+                                getattr(owner.op, 'name', None) == owner_name):
+                            must_ignore = True
+                            break
+            if must_ignore:
+                continue
+            conditional_on.add(parent)
+        else:
+            parent_owner = getattr(parent, 'owner', None)
+            queue.extend(getattr(parent_owner, 'inputs', []))
+    if not conditional_on:
+        return None
+    return conditional_on
+
+
+def add_to_dependence_dag(dag, node, return_added_node=False, force=False,
```

Comment:
I don't think you need `return_added_node` here and should always return the node. Where you call it with `return_added_node=False` above you never access the return arguments.